### DISCUSSION TITLE
How would I integrate this library with Wagmi/React?

### DISCUSSION THREAD

**Discussion by zkSoju** [ProjectOpenSea/seaport-js#74] [Q&A]
Hi there!

Curious what would be the best approach to utilizing this library with Wagmi React hooks? (https://github.com/tmm/wagmi/) I was following an active PR about passing a signer to the Seaport constructor, but I couldn't figure out how to expose that in Wagmi. 

I think an option is utilizing this repos utility functions to create a custom hook library with wagmi under the hood.

**Comment by sunwrobert** [comment]
Is there a more specific aspect that you need help figuring out? Not familiar with Wagmi so not sure what exactly is the issue here. Maybe an example use case and how you would want to use it would be best here.

**Reply by zkSoju** [reply]
> Is there a more specific aspect that you need help figuring out? Not familiar with Wagmi so not sure what exactly is the issue here. Maybe an example use case and how you would want to use it would be best here.

Makes more sense now with the providerOrSigner parameter since I'm able to expose the signer object. However, how would you supply `_signTypedData` in this case? Thanks!

```
export type Signer = ethers.Signer & {
  _signTypedData(
    domain: TypedDataDomain,
    types: Record<string, Array<TypedDataField>>,
    value: Record<string, any>
  ): Promise<string>;
};
```

**Reply by sunwrobert** [reply]
Hm, I was under the assumption that signTypedData would be provided for the most part for custom signers. What type of signer are you providing and how would it normally sign typed data?

**Reply by zkSoju** [reply]
> Hm, I was under the assumption that signTypedData would be provided for the most part for custom signers. What type of signer are you providing and how would it normally sign typed data?

The signer is supplied as an `ethers.Signer` by wagmi (https://wagmi.sh/docs/hooks/useSigner), but it doesn't seem to supply `_signTypedData` that's shown here: https://docs.ethers.io/v5/api/signer/#Signer-signTypedData

**Reply by sunwrobert** [reply]
Are you using it in the context of the browser? It's likely that it actually has signTypedData provided. I would try ignoring the type there and seeing if it works, and then maybe we can adjust the type accordingly in seaport-js

**Reply by zkSoju** [reply]
Hi, is there an example use case of this?

**Reply by BravoNatalie** [reply]
I'm with the same problem, and it works when you ignore the error like this:
```
 //@ts-ignore
const seaport = new Seaport(signerFromWagmi)
```

**Reply by zkSoju** [reply]
> I'm with the same problem, and it works when you ignore the error like this:
> 
> ```
>  //@ts-ignore
> const seaport = new Seaport(signerFromWagmi)
> ```

If you're having issues still, I followed this implementation - it works well:
https://github.com/jeff0723/3card/blob/main/packages/web/src/providers/SeaportProvider.tsx

It works well, just make sure to try catch `executeAllActions` because there are missing signer errors when transactions get denied.

**Reply by bymi15** [reply]
For anyone using more recent Wagmi versions, you can follow https://wagmi.sh/react/guides/ethers#connector-client-%E2%86%92-signer to get the signer and the above post works well!
